### DISCUSSION TITLE
What's the expected structure of an S3 Bucket?

### DISCUSSION THREAD

**Discussion by dontlaugh** [fluxcd/flux2#3260] [Q&A]
I'm familiar with using Flux with Git.

Normally I favor a [monorepo structure with kustomize overlays](https://fluxcd.io/flux/guides/repository-structure/#monorepo) for my different environments.

How do I translate this into an S3 bucket?

Do I literally just lift and shift the file structure of my repos into S3 paths? Where each yaml file becomes an object?

**Comment by uhari03** [comment]
@dontlaugh I also have the same question. My YAMLs in S3 are currently zipped up and I'm wondering if Flux CD will unzip when it processes them.

Did you find an answer to your question?
